source/base/quadrature.cc

```diff
@@ -109,7 +114,8 @@ Quadrature<dim>::Quadrature (const SubQuadrature &q1,
                              const Quadrature<1> &q2)
   :
   quadrature_points (q1.size() * q2.size()),
-  weights (q1.size() * q2.size())
+  weights (q1.size() * q2.size()),
+  is_tensor_product_flag (true)
```

Comment:
Isn't this only true if `q1` is a tensor product formula. And without the suggested generalization you would also need to check that the generating factors of `q1` are the same as `q2`.